admin/admin.go

```diff
@@ -0,0 +1,120 @@
+package admin
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"net/rpc"
+	"os"
+	"runtime"
+	"runtime/pprof"
+	"strings"
+)
+
+const sockAddr = "/tmp/rudder-server.sock"
```

Comment:
Earlier we had an issue reported by one of our users, IIRC some specific Suse/Redhat version, `/tmp` directory wasn't available. So the server kept crashing.

That's when we added a temp variable RUDDER_TMPDIR. Can we also do the same here? Use RUDDER_TMPDIR if present or fallback to `/tmp`

Reply:
@snarkychef Won't it be tricky on rudder-cli side because this path is hard-coded there? May be it shouldn't be hardcoded there either?

Reply:
Yeah. As RUDDER_TMPDIR is environment variable anyways, it's feasible to expect the variable to be available for rudder-cli as well.

Reply:
At least in docker container it will be available. If someone deploys rudder-server by themselves using binary and supervisor etc. It might fail.